### PR TITLE
Clean up

### PR DESCRIPTION
**Pull request by jacobweinstock**
## Description

Looking to push through needed/desired code changes from PR #139 and #140. Remove insecure downloading of container registry certificate. Add the ability to specify insecure registries for use in the docker container engine config. 

## Why is this needed

Fixes: #

## Checklist:

I have:

- [ ] updated the documentation and/or roadmap (if required)
- [ ] added unit or e2e tests
- [ ] provided instructions on how to upgrade

### CODE REVIEW

**Comment by jacobweinstock** [comment]
Hey @thebsdbox , would you mind checking this out when you have a moment. Please and thank you!

File: hook-docker/main.go [hunk 1]
```diff
@@ -38,28 +34,23 @@ func main() {
...
+	}
+	path := "/etc/docker"
+	// Create the directory for the docker config
+	err = os.MkdirAll(path, os.ModeDir)
+	if err != nil {
+		panic(err)
 	}
-	if err := d.writeToDisk("/etc/docker/daemon.json"); err != nil {
+	if err := d.writeToDisk(filepath.Join(path, "daemon.json")); err != nil {
 		fmt.Println("Failed to write docker config:", err)
```
Comment (chrisdoherty4) [inline hook-docker/main.go:54]: Shouldn't we have a `return` after this?

Comment (jacobweinstock) [reply hook-docker/main.go:54]: ah good catch, thanks! will update

**Comment by jacobweinstock** [comment]
@mergify queue

**Pull request merged**